### PR TITLE
feat(teams): implement teams cmd

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
implement teams cmd in up-cli

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

tested the implementation from up-sdk-go with an implementation in up-cli 

added in go.mod

cat go.mod
```
[...]
replace github.com/upbound/up-sdk-go => ../up-sdk-go
```

```
up team create test-implementation -a upboundcare
upboundcare/test-implementation created
```

image

```
up team get test-implementation -a upboundcare   
NAME                  ID                                     CREATED
test-implementation   f68ca5fd-4a83-40bb-a620-3db3da3c9109   17s
```

```
up team list -a upboundcare
NAME                                       ID                                     CREATED
test123                                    5e8e8dab-3598-4003-a5ba-34a12859d60f   23m
test-implementation                        f68ca5fd-4a83-40bb-a620-3db3da3c9109   31s
```

```
up team delete test-implementation  -a upboundcare
Are you sure you want to delete this team? [y/n]: y
Deleting team upboundcare/test-implementation. This cannot be undone.
upboundcare/test-implementation deleted
```

```
up team list -a upboundcare                       
NAME                                       ID                                     CREATED
test123                                    5e8e8dab-3598-4003-a5ba-34a12859d60f   23m
```

### CODE REVIEW

File: cmd/up/team/delete.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		return nil
+	}
+
+	confirm, err := c.prompter.Prompt("Are you sure you want to delete this team? [y/n]", false)
+	if err != nil {
+		return err
+	}
+
+	if input.InputYes(confirm) {
+		p.Printfln("Deleting team %s/%s. This cannot be undone.", upCtx.Account, c.Name)
```
Comment (RedbackThomson) [inline cmd/up/team/delete.go:56]: I would move "This cannot be undone" to the confirmation prompt

File: cmd/up/team/delete.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	a, err := ac.Get(ctx, upCtx.Account)
+	if err != nil {
+		return err
+	}
+
+	if a.Account.Type != accounts.AccountOrganization {
+		return errors.New(errUserAccount)
+	}
+
+	rs, err := oc.ListTeams(ctx, a.Organization.ID)
```
Comment (RedbackThomson) [inline cmd/up/team/delete.go:83]: Does this paginate properly? If the name isn't in the first page then this method will 404

File: cmd/up/team/get.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return err
+	}
+
+	for _, r := range rs {
+		if r.Name == c.Name {
+			return printer.Print(r, fieldNames, extractFields)
+		}
+	}
+	return errors.New("no team named \"" + c.Name + "\"")
```
Comment (RedbackThomson) [inline cmd/up/team/get.go:68]: nit: use `fmt.Errorf`

File: cmd/up/team/delete.go (hunk 2 above)
Comment (haarchri) [reply cmd/up/team/delete.go:83]: thats today true for a lot of more requests like robots or ?

Comment (RedbackThomson) [reply cmd/up/team/delete.go:83]: I'm not sure all of the APIs even support pagination. But hopefully the new ones do 🤷‍♂️

Comment (haarchri) [reply cmd/up/team/delete.go:83]: Thats also an old API oet me try but first try was without

File: cmd/up/team/delete.go (hunk 1 above)
Comment (haarchri) [reply cmd/up/team/delete.go]: done

File: cmd/up/team/delete.go (hunk 2 above)
Comment (haarchri) [reply cmd/up/team/delete.go:83]: API not supports paginate today

File: cmd/up/team/get.go (hunk 3 above)
Comment (haarchri) [reply cmd/up/team/get.go]: done

File: cmd/up/team/delete.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	if err != nil {
+		return err
+	}
+
+	if input.InputYes(confirm) {
+		p.Printfln("Deleting team %s/%s. This cannot be undone.", upCtx.Account, c.Name)
+		return nil
+	}
+
+	return fmt.Errorf("operation canceled")
```
[the remark this answers is not in the document]
Comment (haarchri) [reply cmd/up/team/delete.go:60]: when returning nil here - the team will be deleted but we want to cancel the operation here

**Comment by haarchri** [comment]
@RedbackThomson can you review again ?

**Review by RedbackThomson — APPROVED** [review]

**Pull request merged**